cpp/tests/prims/mg_update_frontier_v_push_if_out_nbr.cu

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utilities/high_res_clock.h>
+#include <utilities/base_fixture.hpp>
+#include <utilities/device_comm_wrapper.hpp>
+#include <utilities/test_graphs.hpp>
+#include <utilities/test_utilities.hpp>
+#include <utilities/thrust_wrapper.hpp>
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/partition_manager.hpp>
+#include <cugraph/utilities/dataframe_buffer.cuh>
+
+#include <cuco/detail/hash_functions.cuh>
+#include <cugraph/graph_view.hpp>
+#include <cugraph/matrix_partition_view.hpp>
+#include <cugraph/prims/copy_to_adj_matrix_row_col.cuh>
+#include <cugraph/prims/row_col_properties.cuh>
+#include <cugraph/prims/update_frontier_v_push_if_out_nbr.cuh>
+#include <cugraph/prims/vertex_frontier.cuh>
+
+#include <thrust/copy.h>
+#include <thrust/count.h>
+#include <thrust/equal.h>
+#include <thrust/iterator/permutation_iterator.h>
+#include <raft/comms/comms.hpp>
+#include <raft/comms/mpi_comms.hpp>
+#include <raft/handle.hpp>
+#include <rmm/device_scalar.hpp>
+#include <rmm/device_uvector.hpp>
+#include <sstream>
+
+#include <gtest/gtest.h>
+
+#include <random>
+
+template <typename TupleType, typename T, std::size_t... Is>
+__device__ __host__ auto make_type_casted_tuple_from_scalar(T val, std::index_sequence<Is...>)
+{
+  return thrust::make_tuple(
+    static_cast<typename thrust::tuple_element<Is, TupleType>::type>(val)...);
+}
+
+template <typename property_t, typename T>
+__device__ __host__ auto make_property_value(T val)
+{
+  property_t ret{};
+  if constexpr (cugraph::is_thrust_tuple_of_arithmetic<property_t>::value) {
+    ret = make_type_casted_tuple_from_scalar<property_t>(
+      val, std::make_index_sequence<thrust::tuple_size<property_t>::value>{});
+  } else {
+    ret = static_cast<property_t>(val);
+  }
+  return ret;
+}
+
+template <typename vertex_t, typename property_t>
+struct property_transform_t {
+  int mod{};
+
+  constexpr __device__ property_t operator()(vertex_t const v) const
+  {
+    static_assert(cugraph::is_thrust_tuple_of_arithmetic<property_t>::value ||
+                  std::is_arithmetic_v<property_t>);
+    cuco::detail::MurmurHash3_32<vertex_t> hash_func{};
+    return make_property_value<property_t>(hash_func(v) % mod);
+  }
+};
+
+struct Prims_Usecase {
+  bool check_correctness{true};
+};
+
+template <typename input_usecase_t>
+class Tests_MG_update_frontier_v_push_if_out_nbr
```

Comment:
We have been naming this class with camel case primitive names (e.g. Tests_MG_TransformReduceV) so better be consistent. I am OK with the current name but in that case, we need to fix elsewhere as well to be consistent.